.circleci/config.yml

```diff
@@ -17,101 +19,114 @@ workflows:
-            VERSION=$(grep 'VERSION=\"[0-9\.]*\"' codecov | cut -d'"' -f2);
-            SHAVERSION=$(shasum -v);
-            echo "Using CodeCov version '$VERSION'"
-            echo "Using shasum '$SHAVERSION'"
-            for i in 1 256 512
-            do
-              shasum -a $i -c --ignore-missing <(curl -s "https://raw.githubusercontent.com/codecov/codecov-bash/${VERSION}/SHA${i}SUM") ||
-              shasum -a $i -c <(curl -s "https://raw.githubusercontent.com/codecov/codecov-bash/${VERSION}/SHA${i}SUM" | grep -w "codecov")
-            done
-            bash ./codecov -v -X s3 -c -D "./ci/unit-test/build/reports" -F unit-tests
-
-      - store_artifacts:
-          path: ci/unit-test/build/reports
+            make unit-test-coverage
 
       - store_test_results:
-          path: ci/unit-test/build/test-results
+          path: code/edgebridge/build/test-results/testPhoneDebugUnitTest
+
+      # Upload code coverage report to Codecov
+      - codecov/upload:
+          file: code/edgebridge/build/reports/coverage/test/phone/debug/report.xml
+          flags: unit-tests
+      
+      - android/save-gradle-cache
+
+      - android/save-build-cache
 
   functional-test:
     executor:
       name: android/android-machine
-      resource-class: large
+      resource-class: medium
+      tag: 2024.01.1
     steps:
       - checkout
-      - android/start-emulator-and-run-tests:
-          #   It should match the name seen in the "sdkmanager --list" output
+
+      - android/restore-build-cache
+
+      - android/create-avd:
+          avd-name: myavd
+          install: true
           system-image: system-images;android-29;default;x86
-          #   The command to be run, while waiting for emulator startup
-          post-emulator-launch-assemble-command: make ci-build
-          #   The test command
-          test-command: make ci-functional-test
-      
-      # code coverage
+
+      - android/start-emulator:
+          avd-name: myavd
+          no-window: true
+          post-emulator-launch-assemble-command: ""
+          restore-gradle-cache-prefix: v1a
+
       - run:
-          name: Upload Code Coverage Report
+          name: Run Functional tests
           command: |
-            curl -s https://codecov.io/bash > codecov;
-            VERSION=$(grep 'VERSION=\"[0-9\.]*\"' codecov | cut -d'"' -f2);
-            SHAVERSION=$(shasum -v);
-            echo "Using CodeCov version '$VERSION'"
-            echo "Using shasum '$SHAVERSION'"
-            for i in 1 256 512
-            do
-              shasum -a $i -c --ignore-missing <(curl -s "https://raw.githubusercontent.com/codecov/codecov-bash/${VERSION}/SHA${i}SUM") ||
-              shasum -a $i -c <(curl -s "https://raw.githubusercontent.com/codecov/codecov-bash/${VERSION}/SHA${i}SUM" | grep -w "codecov")
-            done
-            bash ./codecov -v -X s3 -c -D "./ci/functional-test/build/reports" -F functional-tests
+            make functional-test-coverage
```

Comment:
I know the User Profile project calls these individual commands, but the android orb has a wrapper which creates the avd, starts the emulator, and runs the tests all in one command. We've been using it in the Edge project.
```
- android/start-emulator-and-run-tests:
          #   It should match the name seen in the "sdkmanager --list" output
          system-image: system-images;android-29;default;x86
          #   The command to be run, while waiting for emulator startup
          post-emulator-launch-assemble-command: make assemble-phone
          #   The test command
          test-command: make functional-test-coverage
          restore-gradle-cache-prefix: << parameters.gradle-cache-prefix >>
```